File: pp_lap.py
import math
def ham_lap(x):
    return math.sqrt(1-4*math.sin(x)) 
 
def pp_lap(x, q, ep):
    cnt = 1
    while abs(ham_lap(x) - x) > (ep * (1 - q) / q):
        print(f"x{cnt}              {x:.9f}")
        x = ham_lap(x)
        print(x)
        cnt += 1

def main():
    x = float(input("NHAP GIA TRI X BAT KI TRONG KHOANG CACH LY : "))
    q = float(input("\nNHAP GIA TRI Q-MAX(ABS(F_DERIVATIVE1)) TREN KHOANG CACH LY : "))
    ep = float(input("\nNHAP GIA TRI EPXILON : "))
    pp_lap(x, q, ep)

if __name__ == "__main__":
    main()
    

File: pp_lap.cpp
#include<iostream>
#include<math.h>
#include<iomanip>
using namespace std;
double ham_lap(double x){
    return (-pow(x,5)+17)/25;
}
void pp_lap(double x,double q,double ep){
    int cnt=1;
    while(abs(ham_lap(x)-x)>(ep*(1-q)/q)){
        cout<<"x"<<cnt<<"              ";
        cout<<setprecision(9)<<x;
        cout<<endl;
        x=ham_lap(x);
        cnt++;
    }
}
int main(){
    double x,q,ep;
    cout<<"NHAP GIA TRI X BAT KI TRONG KHOANG CACH LY : ";cin>>x;
    cout<<"\nNHAP GIA TRI Q-MAX(ABS(F_DERIVATIVE1)) TREN KHOANG CACH LY : ";cin>>q;
    cout<<"\nNHAP GIA TRI EPXILON : ";cin>>ep;
    pp_lap(x,q,ep);
    return 0;
}